hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieCompactor.java

```diff
@@ -34,17 +32,31 @@
 
   /**
    * Compact the delta files with the data files
+   *
+   * @deprecated : Will be removed in next PR
    */
+  @Deprecated
   JavaRDD<WriteStatus> compact(JavaSparkContext jsc, final HoodieWriteConfig config,
       HoodieTable hoodieTable, String compactionCommitTime) throws Exception;
 
+  /**
+   * Generate workload for scheduling compaction
+   *
+   * @param jsc                  Spark Context
+   * @param hoodieTable          Hoodie Table
+   * @param config               Hoodie Write Configuration
+   * @param compactionCommitTime scheduled compaction commit time
+   * @return Compaction Workload
+   * @throws IOException when encountering errors
+   */
+  HoodieCompactionWorkload generateCompactionWorkload(JavaSparkContext jsc,
+      HoodieTable hoodieTable, HoodieWriteConfig config, String compactionCommitTime)
+      throws IOException;
 
-  // Helper methods
-  default String startCompactionCommit(HoodieTable hoodieTable) {
-    String commitTime = HoodieActiveTimeline.COMMIT_FORMATTER.format(new Date());
-    HoodieActiveTimeline activeTimeline = hoodieTable.getActiveTimeline();
-    activeTimeline
-        .createInflight(new HoodieInstant(true, HoodieTimeline.COMMIT_ACTION, commitTime));
-    return commitTime;
-  }
-}
+  /**
+   * Execute compaction operations and report back status
+   */
+  JavaRDD<WriteStatus> executeCompaction(JavaSparkContext jsc,
```

Comment:
Rename this to compact() and get rid of the other compact() method..

Reply:
Done